### PR TITLE
Adding producer blow up test

### PR DESCRIPTION
**Pull request by nhibberd**
@charleso

### CODE REVIEW

File: test/Test/Twine/Parallel.hs [hunk 1]
```diff
@@ -63,6 +63,18 @@ prop_worker_blow_up = forAll (choose (1, 1000 :: Int)) $ \n -> testIO $ do
       False
   pure $ z === True
 
+prop_worker_blow_up_producer = testIO $ do
+  let pro = const . throwM . userError $ "producer"
+      work = const $ pure ()
+
+  r <- runEitherT $ consume pro 1 work
+  z <- pure $ case r of
```
Comment (charleso) [inline test/Test/Twine/Parallel.hs]: Would just remove `z <-` and the next line, that should work?

**Comment by charleso** [comment]
:+1:

File: test/Test/Twine/Parallel.hs (hunk 1 above)
Comment (nhibberd) [inline test/Test/Twine/Parallel.hs]: And leave the type as Bool over Property? or

Comment (charleso) [inline test/Test/Twine/Parallel.hs]: > And leave the type as Bool over Property? or

`Bool` is `Testable` so you should be fine

**Pull request merged**